src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -389,7 +388,7 @@ private bool ShouldBeInFullSyncMode(Snapshot best)
             bool notInBeaconModes = !best.IsInAnyBeaconMode;
             bool desiredPeerKnown = AnyDesiredPeerKnown(best);
             bool postPivotPeerAvailable = AnyPostPivotPeerKnown(best.Peer.Block);
-            bool hasFastSyncBeenActive = best.Header >= _pivotNumber;
+            bool hasFastSyncBeenActive = _syncConfig.FastSync == false || best.Header >= _pivotNumber;
```

Comment:
hmmm... if we don't have fastSync at all... we're returning hasFastSyncBeenActive = true

Reply:
Yes, but if its not set, full sync mode will not triggered at all.